src/App.js

```diff
@@ -1,14 +1,142 @@
 import React from 'react';
 
+import Header from './Header';
+import Search from './Search';
+import Movies from './Movies';
+import Loading from './elements/Loading';
+import BackToTop from './elements/BackToTop';
+
+
 class App extends React.Component {
-  constructor(){
+  constructor() {
     super();
+    this.state = {
+      search: [],
+      searchTerm: "",
+      page: 2,
+      watchList: localStorage.getItem("watchList") ? localStorage.getItem("watchList") : [],
+      watchListSection: false
+    }
+    this.receiverSearch = this.receiverSearch.bind(this);
+    this.receiverWatchList = this.receiverWatchList.bind(this);
+    this.receiverDisplayWatchlist = this.receiverDisplayWatchlist.bind(this);
+    this.handleScroll = this.handleScroll.bind(this);
+  }
+
+  handleScroll() {
+    const windowHeight = "innerHeight" in window ? window.innerHeight : document.documentElement.offsetHeight;
+    const body = document.body;
+    const html = document.documentElement;
+    const docHeight = Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);
+    const windowBottom = windowHeight + window.pageYOffset;
+    const backToTop = body.querySelector("#back-to-top");
+
+    const loading = document.querySelector('#search-results__loading');
+    const OMBbAPIKey = "aba7af8e";
+    const query = 'http://www.omdbapi.com/?s=' + this.state.searchTerm + '&apikey=' + OMBbAPIKey + '&page=' + this.state.page;
+
+    // Back to top button
+    if (body.scrollTop > 300 || html.scrollTop > 300) {
```

Comment:
Rather than manipulating the DOM by adding classes, it would be better to set property in state based on scrollTop and set the class using JSX